### PR TITLE
tidy up tests, consistent reference to  and not  or

### PR DESCRIPTION
**Pull request by amaarora**
This PR attempts to tidy up tests in `TfmdLists`. There were a lot of overlapping tests for the same thing, this PR attempts to fix that. 
Also, updates names to make more sense in the documentation. 
After spending a lot of time and understanding `TfmdLists`, few new tests have also been added that would help the user reading the documentation. 

As an example:
```
test_eq(tl.train, tl.subset(0))
test_eq(tl.valid, tl.subset(1))
test_eq(tl.train.items, items[splits[0]])
test_eq(tl.valid.items, items[splits[1]])
```

Also, in some funcs of `TfmdList`, obj was referred to as `o` while elsewhere as `x`, this PR attempts to fix that for a part of the notebook involving `TfmdList`s.

If this PR get's merged, I will create a new one, to tidy up tests for `dsrc` depending on feedback :)

### CODE REVIEW

**Comment by amaarora** [comment]
I have tried to make sure that we do not lose any tests, but assumed that if a test passes once of a `tl`, the same test will pass again on a diff `tl`, to reduce overlapping tests for the same thing. 

Also, 
```
FilteredBase.train,FilteredBase.valid = add_props(lambda i,x: x.subset(i), 2)
```
In `add_props`, n=2 by default there `2` is removed from `add_props(lambda i,x: x.subset(i), 2)` as its not needed. 

Documentation for `subset` has been updated to "New `TfmdList` with same tfms that only includes items in `i`th split".

These are the entirety of changes proposed in this PR.

**Comment by amaarora** [comment]
Always happy to receive feedback and make updates if needed :)

File: dev/05_data_core.ipynb [hunk 1]
```diff
@@ -669,7 +669,7 @@
     "               for i,(b,s,dk) in enumerate(zip(bss,shuffles,dl_kwargs))]\n",
```
Comment (sgugger) [inline dev/05_data_core.ipynb]: <p>Please keep the names private for classes and functions only used in tests, as we don't want to pollute the state too much :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastai_dev/pull/293/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

**Comment by sgugger** [comment]
Thanks a lot for this clean-up! I just have one comment on names you changed, but other than that it looks good!

**Comment by amaarora** [comment]
Thanks @sgugger for the feedback 

I have updated names of functions and classes used in only tests with `_` at the start :)

**Comment by amaarora** [comment]
PS: I sent you a review request by mistake, it is like the red button I wanted to press to see what it does :)

**Comment by sgugger** [comment]
Thanks!

**Pull request merged**